### ISSUE TITLE
Slow first .cuda() call

### ISSUE THREAD

**Issue by W4ngatang** [pytorch/pytorch#5302]
I upgraded to v0.3.1 using conda and have since experienced really slow performance the first time I call .cuda(). I tried solutions listed [here](https://discuss.pytorch.org/t/model-cuda-takes-long-time/102/20) and [here](https://discuss.pytorch.org/t/model-cuda-takes-long-time/102/20) but no luck. I then tried installing the latest version from source but still experienced the slowdown. Any help would be hugely appreciated!

Info
- OS: CentOS 7.4.1708
- PyTorch version: tried 0.3.1 (conda) and 0.4.0a0+6279367 (source)
- How you installed PyTorch (conda, pip, source): conda, source
- Python version: 3.6.3
- CUDA/cuDNN version: 8.0.61, 7.0.2
- GPU models and configuration: 1080ti
- GCC version (if compiling from source): 7.2.0

**Comment by SsnL** [comment]
The cuda context is lazily initialized, so it's expected that the first cuda() call is slower.

**Comment by W4ngatang** [comment]
With the update, the delay is on the order of minutes to cuda a model,
whereas previously it was just a few seconds, so it seems like something is
wrong

**Comment by W4ngatang** [comment]
[build_log.log](https://github.com/pytorch/pytorch/files/1741280/build_log.log)
[nvidia-smi.log](https://github.com/pytorch/pytorch/files/1741281/nvidia-smi.log)

Log files from calling '> python setup.py build develop' and '> nvidia-smi'

**Comment by W4ngatang** [comment]
Ended up working itself out...probably a hardware issue

**Comment by colesbury** [comment]
CUDA JIT compiles all the kernels in the shared library when it's initialized. This can take a few minutes the first time. It caches the results (usually in `~/.nv/ComputeCache/`) so subsequent sessions should be much quicker.

**Comment by soumith** [comment]
@colesbury our binary and source builds dont make CUDA JIT-compile anything. We ship SASS for all supported GPU architectures.

@W4ngatang I'm so glad it worked itself out, we couldn't think of what was going wrong

**Comment by colesbury** [comment]
If you encounter the issue again, run the CUDA initialization under strace:

```
strace python -c "import torch; torch.cuda.FloatTensor(1)"
```

or to log to a file

```
strace python -c "import torch; torch.cuda.FloatTensor(1)" 2>&1 | tee strace.log
```

**Comment by Twittadrock** [comment]
I have the same issue. It seems as if pytorch is looping the whole memory. 
I attached the strace of:
strace python -c "import torch; torch.cuda.FloatTensor(1)" 2>&1 | tee strace.log
Anyway, the issue seems to be fixed with 0.3.0.post4, but might be helpful in case of a regression.

Affected Version:
------------------
Python: (3, 6, 4)
CuDNN: 7005
CUDA: 8.0.61
Torch: 0.3.0
Numpy: 1.14.0

[strace.zip](https://github.com/pytorch/pytorch/files/1745793/strace.zip)

**Comment by piotrm0** [comment]
Am on 1.4.0 and I have this issue.
